mirrord/layer/src/file/hooks.rs

```diff
@@ -322,7 +325,9 @@ pub(crate) unsafe extern "C" fn pread_detour(
             //
             // Callers can check for EOF by using `ferror`.
             if read_amount > 0 {
-                let bytes_slice = &bytes[0..fixed_read as usize];
+                let bytes_slice = bytes
+                    .get(..fixed_read as usize)
+                    .expect("read_amount exceeds bytes length in ReadFileResponse");
```

Comment:
I think it's better we propagate errors? It's probably pretty fatal but we would get nicer error message I believe?

Reply:
Let's chat about on Discord